core/src/main/java/io/micronaut/core/util/NativeImageUtils.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.core.util;
+
+/**
+ * Utility class to retrieve information about the context in which code gets executed.
+ * Partial fork of {@code org.graalvm.nativeimage.ImageInfo} to avoid a dependency on {@code org.graalvm.sdk:nativeimage}.
+ *
+ * @since 4.8.0
+ */
+public final class NativeImageUtils {
```

Comment:
I think we should keep it internal

Reply:
resolved via https://github.com/micronaut-projects/micronaut-core/pull/11388/commits/927ee3398ed46302095bdfebd7059d57577ab96c